### ISSUE TITLE
Clicks and crackle noise on Prologue 8

### ISSUE THREAD

**Issue by Sebo1971** [dukesrg/logue-fx#1]
Hi Oleg:
First of all thank you for doing all this FXs, I loved Gator and Hyper.
I tried the precompiled Gator, Hyper, Crusher and Looper in my Prologue 8.
Gator and Looper works well but Hyper and Crusher have crackle noise and random clicks all time.
I tried them as ModFX, as DelFx and as RevFX and they behave the same in all slots.
I don't know if it have something to do with the fact they are compiled for the NTS-1, I used oscillators and FXs compiled for the Minilogue XD without problems. I can't compile them in the Prologue format myself (I'm trying to learn how to use github and I have a basic knowledge of C, but all my attempts to compile a Logue SDK project where unsuscessfull).
Can you check for this issues, I will love to use Hyper. I could test them in my Prologue 8 and give you feedback.
Thank You.

**Comment by dukesrg** [comment]
@Sebo1971 I added Crusher and Hyper precompiled for Prologue in the release section. It should be the same, but anyway it is better to check.
Hyper is clicking by design because of simple pitch shift simulation. It might be unnoticeable with sawtooth wave input.
Crusher is noisy on NTS-1 with audio input with low bit depth set, I have no idea what else can produce clicks :(

**Comment by Sebo1971** [comment]
Hi Oleg: Thank you for your response and for taking the time to precompile the effects!
Gator was working well before (may be I wasn't clear about it).
Hyper still having the clicks and crackles as before... As you said if I use sawtooths or other harmonically rich waves, the crackle gets masked but the effect is so great on wavetables (it gives me a kind of Wavestation feel that I love) that I want to use it with waves that are not so rich in harmonics...
Anyway, thank you!

**Comment by dukesrg** [comment]
@Sebo1971 added the Crusher for Prologue, please test it.

I also created a supersaw oscillator with pretty the same algorithm as Hyper. It is clean but surprisingly Hyper gives more richer sound.

**Comment by Sebo1971** [comment]
Hi Oleg, Crusher still the same... I noted that the noise only appears when I reduce the bitdepth, with the knob at 0% there is no noise (I can set the sampling frequency to wathever I want), but as soon as I turn a little bit the knob the crackle starts and increases as I reduce the bitdepth.
I tried turning down all three oscillators and closing the filter but the cracle persist, I think it is produced by analog noise or the jitter/noise of the ADC, I don't know, just guessing. May be a simple noise gate before the actual bitcrush algorithm will do the trick...

Hyper is so great, without the crackle will be perfect, because as is I can only use it in rich harmonic waves.

Thanks a lot for your efforts!!!

**Comment by dukesrg** [comment]
@Sebo1971 sounds very similar to NTS-1 and audio input routed through Crusher.
AFAIK Prologue does not have audio input, but probably there is ADC for analog oscillators passing through the effect.
Can you try to disable/attenuate analog oscillators and get input form the multi engine?

**Comment by Sebo1971** [comment]
Hi Oleg, that's what I did (I setted all 3 oscillators to 0). Take in account that the digital oscillator is converted to analog in order to be processed by the analog filter and/or analog VCA, so there is no digital path to the effects. It always goes thought the ADC.

**Comment by dukesrg** [comment]
@Sebo1971 can you please record and share the output of the silence processed with only Crusher while increasing the bit depth reduction parameter?  I'll compare with what I can get from NTS-1.
NTS-1 output is noisy, even LED refresh is interfering and audio in ADC for sure.

**Comment by Sebo1971** [comment]
Hi Oleg:
Here it is a recording of my Prologue 8 using Crusher. I'm not playing any note, and using only Crusher as FX. I started with bitdepth at 0% and slowly increasing it to 100% then back to 0%.
Hope this help.
https://drive.google.com/open?id=1vUbkjUwXBzUhsc0-7-0YCYGzXf-ygkCz

**Comment by dukesrg** [comment]
@Sebo1971 Check updated Crusher version in releases. Now it also could works like noise gate on higher bit crush setting :)

**Comment by Sebo1971** [comment]
HI, I tried the new version of Crusher and the crackle noise is gone!!! Great!
But I noticed that the bitcrush control works from 0% to 50%, from 50% to 100% I have no sound at all. I don't know if it is on purpose or not, but I think it will be better that the bitcrush works on the full range of the knob...
Thank you!

**Comment by dukesrg** [comment]
@Sebo1971 Bit depth reduction works Ina full range while signal typically sustained below +/-0.5 level.
Start the note, find the bit depth threshold where sound just disappeared. Trigger the note again and you'll get a bit of a sound at attack stage where it is louder.
The noisy version was rounding to +/- infinity, so it always produces sound but any noise is also expanded. The latest version is rounding just like Waves does with bitcrush. But Waves adds white noise proportionally. It should be fine for an oscillator but it's not for the effect.

**Comment by Sebo1971** [comment]
I tested Crusher again and I think I found why it behaves like that.
The first test I did it with just one oscillator (at max level) and sustain to max, and as I said Crusher get silent with bitdepth at about 50%.
Now I tried with all three oscillators at max level, drive at max and sustain at max,  Crusher gets silent at about 70% of bitdepth. If I play 4 or 5 notes of polyphony with those settings I can rise the bitdepth knob to 90%. I think if I play full 8 notes I could use the bitdepth up to 100%, but to be honest playing more than 3 notes with drive at full and bitdepth reduction at full is just plain noise, jaja...
So I think the difference with NTS-1 is in that the level at the input of the ADC is low to avoid digital clipping when you play polyphonic patches.
Great work!!!
Thank you!!!
